### PR TITLE
GCI-1576 - Environment Variable check prior to application start

### PR DESCRIPTION
**Pull request by caismith-ch**
- Prevent application starting if required environment variables are not present
- Add tests
- Add additional `gitignore` value for log files
- Update README

Resolves: GCI-1576

### CODE REVIEW

File: src/test/java/uk/gov/companieshouse/chd/order/api/ChdOrderApiApplicationTests.java [hunk 1]
```diff
@@ -1,13 +1,195 @@
...
+    private static final String CHPRD_HANDCSR = "CHPRD_HANDCSR";
+    private static final String CHPRD_STATUS = "CHPRD_STATUS";
+    private static final String CHPRD_FLAGS = "CHPRD_FLAGS";
+    private static final String CHPRD_LANGUAGE = "CHPRD_LANGUAGE";
+    private static final String CHPRD_DELIVMETH = "CHPRD_DELIVMETH";
+    private static final String CHPRD_DELIVLOCATION = "CHPRD_DELIVLOCATION";
+
+    @Test
+    void checkEnvironmentVariablesAllPresentReturnsTrue() {
+        setAllEnvironmentVariables();
```
Comment (nabeelamirch) [inline src/test/java/uk/gov/companieshouse/chd/order/api/ChdOrderApiApplicationTests.java:31]: You could probably put this in a @BeforeEach

Comment (caismith-ch) [reply src/test/java/uk/gov/companieshouse/chd/order/api/ChdOrderApiApplicationTests.java:31]: Good shout 👍

File: src/test/java/uk/gov/companieshouse/chd/order/api/ChdOrderApiApplicationTests.java [hunk 2]
```diff
@@ -1,13 +1,195 @@
...
+
+        boolean isPresent = ChdOrderApiApplication.checkEnvironmentVariables();
+        assertTrue(isPresent);
+
+        clearEnvironmentVariables();
```
Comment (nabeelamirch) [inline src/test/java/uk/gov/companieshouse/chd/order/api/ChdOrderApiApplicationTests.java:36]: You could probably put this in a @ Aftereach

Comment (caismith-ch) [reply src/test/java/uk/gov/companieshouse/chd/order/api/ChdOrderApiApplicationTests.java:36]: Another good shout 👍

File: src/test/java/uk/gov/companieshouse/chd/order/api/ChdOrderApiApplicationTests.java [hunk 3]
```diff
@@ -1,13 +1,195 @@
 package uk.gov.companieshouse.chd.order.api;
 
+import org.junit.Rule;
```
Comment (nabeelamirch) [inline src/test/java/uk/gov/companieshouse/chd/order/api/ChdOrderApiApplicationTests.java:3]: You don't have to do this one, but you could also use parameterized tests. https://www.tutorialspoint.com/junit/junit_parameterized_test.htm

**Review by nabeelamirch — DISMISSED** [review]

File: src/test/java/uk/gov/companieshouse/chd/order/api/ChdOrderApiApplicationTests.java (hunk 3 above)
Comment (caismith-ch) [reply src/test/java/uk/gov/companieshouse/chd/order/api/ChdOrderApiApplicationTests.java:3]: Nice didn't know about this, will have to spend some time working it out but a great suggestion to look into for the future

**Review by nabeelamirch — APPROVED** [review]

**Pull request merged**